### ISSUE TITLE
No longer installs on BareMetal K8S

### ISSUE THREAD

**Issue by ReggieCarey** [operator-framework/operator-lifecycle-manager#2306]
Apparently OLM installations now appear to depend on openshift. 

I thought this was meant for any k8s environment. Am I wrong in this assumption?

My attempts to install olm either from the website
```bash
curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.18.2/install.sh | bash -s v0.18.2
```
 or via a script
```bash
curl -L https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.18.2/install.sh -o install.sh
chmod +x install.sh
./install.sh v0.18.2
```
now fail silently.  So, I modified install.sh to see the failure
```bash
...
# This script is for installing OLM from a GitHub release

set -xe
...
```

This results in the following output:
```bash
./install.sh v0.18.2
+ default_base_url=https://github.com/operator-framework/operator-lifecycle-manager/releases/download
+ [[ 1 -lt 1 ]]
+ [[ 1 -gt 2 ]]
+ kubectl get deployment olm-operator -n openshift-operator-lifecycle-manager '-o=jsonpath={.spec}'
```

I'm attempting to install OLM on a K8s instance that currently has no operator infrastructure from OperatorHub.io

**Comment by kevinrizza** [comment]
@ReggieCarey This script does not depend on OCP or any flavor of OpenShift at all. Actually the reverse, that change (which is in error) is intended to just fail the script if you attempt to run it on OpenShift, since they are not compatible. There is a bug in that script which was merged erroneously and without proper test coveage that it does not work in certain terminal contexts however, which was already fixed:

https://github.com/operator-framework/operator-lifecycle-manager/pull/2211

As soon as 0.18.3 is released, the automation will be resolved. Alternatively, you can pull the script from the master branch and run it without issue.

In general, this project is intended for and tested on vanilla K8s clusters, and OpenShift is not a dependency.

**Comment by timflannagan** [comment]
One thing to tack onto Kevin's previous comment (as I was in the middle of replying myself :laughing:) - this is a technically a duplicate of https://github.com/operator-framework/operator-lifecycle-manager/issues/2217, which will be closed once v0.18.3 has been released.
